src/main/java/ukecat/Task.java

```diff
@@ -0,0 +1,87 @@
+package ukecat;
+
+/**
+ * Represents a generic task in the UkeCat application.
+ * Acts as the base class for specific task types like ToDo, Deadline, and Event.
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructs a new Task with the specified description.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a new Task with the specified status, description, and isDone value.
+     *
+     * @param isDone      The status of the task ("0" for not done, "1" for done).
+     * @param description The description of the task.
+     */
+    public Task(String isDone, String description) {
+        this.isDone = isDone.equals("1");
+        this.description = description;
+    }
+
+    /**
+     * Sets the task as done or not done based on the provided MarkType.
+     *
+     * @param markType The type of marking (MARK for done, UNMARK for not done).
+     */
+    public void setDone(MarkType markType) {
+        switch (markType) {
+        case MARK:
+            this.isDone = true;
+            System.out.println("  Nice! I've marked this task as done:");
+            break;
+        case UNMARK:
+            this.isDone = false;
+            System.out.println("  OK, I've marked this task as not done yet:");
+            break;
+        }
+        System.out.println("    " + this);
+    }
+
+    /**
+     * Gets the integer representation of the isDone status (1 for done, 0 for not done).
+     *
+     * @return The integer representation of the isDone status.
+     */
+    public int getIntIsDone() {
+        return isDone ? 1 : 0;
+    }
```

Comment:
I don't like the usage of integers here. Consider using an enum, so that adding a new status (like "in progress") would be cleaner and more readable.